Ecal/src/Ecal/EcalVetoProcessor.cxx

```diff
@@ -1261,6 +1279,44 @@ float EcalVetoProcessor::distPtToLine(TVector3 h1, TVector3 p1, TVector3 p2) {
   return ((h1 - p1).Cross(h1 - p2)).Mag() / (p1 - p2).Mag();
 }
 
+std::vector<float> EcalVetoProcessor::trackProp(const ldmx::Tracks &tracks,
+                                                ldmx::TrackStateType ts_type,
+                                                const std::string &ts_title) {
+  // Vector to hold the new track state variables
+  std::vector<float> newTrackStates;
+
+  // Return if no tracks
+  if (tracks.empty()) return newTrackStates;
+
+  // Otherwise loop on the tracks
+  for (auto &track : tracks) {
+    // Get track state for ts_type
+    auto trk_ts = track.getTrackState(ts_type);
+    // Continue if there's no value
+    if (!trk_ts.has_value()) continue;
+    ldmx::Track::TrackState &TargetState = trk_ts.value();
+
+    // Check that the track state is filled
+    if (TargetState.params.size() < 5) continue;
+
+    float track_state_loc0 = static_cast<float>(TargetState.params[0]);
+    float track_state_loc1 = static_cast<float>(TargetState.params[1]);
+
+    // Store the new track state variables
+    newTrackStates.push_back(track_state_loc0);
+    newTrackStates.push_back(track_state_loc1);
+    // z-position at the ECAL scoring plane
+    newTrackStates.push_back(239.999);
```

Comment:
I fear that this hard-coded number will lead to an obscure bug in the future. Can we link this value to the value that gets passed to `getTrackState`?

Also not sure if we need this many decimal points. Why not `240.`?

Reply:
I share your fear, but the track state does not save the Z position, it just does it at the ECAL SP's position. I wish the parameters in the GMDL could be invoked here. I think it's not 240, so it doesnt overlap with the real ECAL in the GDML. I'm open to ideas for how to not have this hard-coded.

Reply:
Let's leave it hardcoded for now. I think resolving this would require some more serious development of the Tracking infrastructure that no one has the time to do right now. I forgot that the TrackStates are pre-computed at Tracking-time and I feel like the z-locations of the track states that are pre-computed should be hardcoded along with their names.

Reply:
>  hardcoded along with their names.

well it adds to the confusing that in tracking X axis = beam axis (i.e. our Z). But yeah this should be indeed done in tracking. I'll "resolve" this comment now then